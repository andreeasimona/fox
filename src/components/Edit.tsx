import React, { useState, useEffect } from "react";
import { withRouter, useHistory } from "react-router-dom";
import { editPost, getPost } from "api/posts";
import { useHooks, initialState } from "utils/hookForm";
import FormPost from "./FormPost";

const Edit = withRouter(({ match: { params } }) => {
  const { id } = params;
  const { state, disabled, dispatch, onChange, checkDisabled } = useHooks();
  const [error, setError] = useState(false);
  const history = useHistory();

  const onSubmit = async () => {
    try {
      await editPost({ ...state, id });
      history.goBack();
    } catch (error) {
      setError(true);
    }
  };

  const onError = () => {
    dispatch({
      field: "image_url",
      value: "https://image.shutterstock.com/image-vector/blank-avatar-photo-place-holder-600w-1114445501.jpg"
    });
  };

  useEffect(() => {
    const fetchData = async () => {
      try {
        const result = await getPost(id);
        for (var field in result) {
          if (initialState.hasOwnProperty(field)) {
            dispatch({ field, value: result[field] });
          }
        }
      } catch (error) {
        setError(true);
      }
    };
    fetchData();
  }, [id]);

  useEffect(() => {
    checkDisabled();
  }, [state]);

  return (
    <div className="posts">
      <h4>Edit</h4>
      {error ? (
        <div className="alert alert-danger" role="alert">
          An error occurred while your post was submitted. Try again later.
        </div>
      ) : (
        <React.Fragment>
          <label>Tumbnail</label>
          <br />
          <img
            alt="post"
            src={state.image_url}
            className="img-thumbnail"
            height="100"
            width="100"
            onError={onError}
          />
          <FormPost
            post={state}
            onChange={onChange}
            onSubmit={onSubmit}
            disabled={disabled}
          ></FormPost>
        </React.Fragment>
      )}
    </div>
  );
});

export default Edit;
