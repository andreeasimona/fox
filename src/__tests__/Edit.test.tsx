import React from "react";
import { MemoryRouter } from "react-router-dom";
import Edit from "../components/Edit";
import { render } from "@testing-library/react";

test("renders the create form component", () => {
  const wrapper = render(
    <MemoryRouter initialEntries={["/edit/2"]}>
      <Edit />
    </MemoryRouter>
  );
  expect(wrapper).toMatchSnapshot();
});
