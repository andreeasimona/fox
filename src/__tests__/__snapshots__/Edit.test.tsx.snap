// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the create form component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="posts"
      >
        <h4>
          Edit
        </h4>
        <label>
          Tumbnail
        </label>
        <br />
        <img
          alt="post"
          class="img-thumbnail"
          height="100"
          src=""
          width="100"
        />
        <form>
          <div
            class="form-group"
          >
            <label>
              Title
            </label>
            <input
              class="form-control"
              name="title"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label>
              Content
            </label>
            <input
              class="form-control"
              name="content"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label>
              Latitude
            </label>
            <input
              class="form-control"
              name="lat"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label>
              Longitude
            </label>
            <input
              class="form-control"
              name="long"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label>
              Image url
            </label>
            <input
              class="form-control"
              name="image_url"
              value=""
            />
          </div>
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="col-11"
              >
                <button
                  class="btn btn-secondary"
                  type="button"
                >
                  Back
                </button>
              </div>
              <div
                class="col-1"
              >
                <button
                  class="btn btn-primary"
                  disabled=""
                  type="button"
                >
                  Submit
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="posts"
    >
      <h4>
        Edit
      </h4>
      <label>
        Tumbnail
      </label>
      <br />
      <img
        alt="post"
        class="img-thumbnail"
        height="100"
        src=""
        width="100"
      />
      <form>
        <div
          class="form-group"
        >
          <label>
            Title
          </label>
          <input
            class="form-control"
            name="title"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label>
            Content
          </label>
          <input
            class="form-control"
            name="content"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label>
            Latitude
          </label>
          <input
            class="form-control"
            name="lat"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label>
            Longitude
          </label>
          <input
            class="form-control"
            name="long"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label>
            Image url
          </label>
          <input
            class="form-control"
            name="image_url"
            value=""
          />
        </div>
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col-11"
            >
              <button
                class="btn btn-secondary"
                type="button"
              >
                Back
              </button>
            </div>
            <div
              class="col-1"
            >
              <button
                class="btn btn-primary"
                disabled=""
                type="button"
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
